src/riscv/linux/cpuinfo.c

```diff
@@ -0,0 +1,315 @@
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <cpuinfo/log.h>
+#include <linux/api.h>
+#include <riscv/linux/api.h>
+
+/*
+ * Size, in chars, of the on-stack buffer used for parsing lines of /proc/cpuinfo.
+ * This is also the limit on the length of a single line.
+ */
+#define BUFFER_SIZE 2048
+
+static uint32_t parse_processor_number(
+        const char* processor_start,
+        const char* processor_end)
+{
+    const size_t processor_length = (size_t) (processor_end - processor_start);
+
+    if (processor_length == 0) {
+        cpuinfo_log_warning("Processor number in /proc/cpuinfo is ignored: string is empty");
+        return 0;
+    }
+
+    uint32_t processor_number = 0;
+    for (const char* digit_ptr = processor_start; digit_ptr != processor_end; digit_ptr++) {
+        const uint32_t digit = (uint32_t) (*digit_ptr - '0');
+        if (digit > 10) {
+            cpuinfo_log_warning("non-decimal suffix %.*s in /proc/cpuinfo processor number is ignored",
+                                (int) (processor_end - digit_ptr), digit_ptr);
+            break;
+        }
+
+        processor_number = processor_number * 10 + digit;
+    }
```

Comment:
Do we consider non-decimal suffixes not suspect? E.g. This would let something like "processor: +++100" through, which should give us pause, right? We're probably reading garbage and shouldn't assume it means 100. To that end, if we don't do any particular corrective action here, couldn't we replace this function with `strtoul` ?